tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -109,6 +105,35 @@ def transformed_log_prob_fn(*state_parts):
     return tlp + sum(ldj_fn(state_parts, event_ndims))
   return transformed_log_prob_fn
 
+def _make_kernel_stack(kernel):
+  kernel_stack = [kernel]
+  while 'target_log_prob_fn' not in kernel.parameters:
+    if 'inner_kernel' not in kernel.parameters:
+      raise ValueError('"None of the nested `inner_kernel`s contains a '
+                       '`target_log_prob_fn`."')
+    kernel = kernel.inner_kernel
+    kernel_stack.append(kernel)
+  return kernel_stack
+
+def get_target_log_prob(kernel):
```

Comment:
i'd suggest calling this `find_nested_target_log_prob_recursively` which is more descriptive. i am not sure it should be part of the public API. is there a reason it needs to be? i'd prefer not to introduce new API tokens unless strictly necessary. The criterion for adding a new API item should be "does a typical user of this object need to use this functionality?". i don't think this is the case here (please let me know if I'm overlooking something).

Reply:
Kind of like with `SimpleStepSizeAdaptation` wouldn't it be useful for the user to be able to get the deepest `target_log_prob_fn` or edit it willingly without needing to manually go through the `inner_kernel`s?

Reply:
I personally like the get/update pair in naming but completely open to change

Reply:
Final thoughts: leave the get/update function private